crates/ty/src/args.rs

```diff
@@ -107,6 +107,15 @@ pub(crate) struct CheckCommand {
     #[clap(flatten)]
     pub(crate) config: ConfigsArg,
 
+    /// A path to a `ty.toml` configuration file
+    ///
+    /// `pyproject.toml` files are not accepted.
+    /// When provided, this file will be used in place of any discovered configuration (including user-level configuration).
+    /// ty will skip project discovery and default to the current working directory.
+    /// Paths are anchored at the current working directory.
```

Comment:
This is probably a bit more detailed than necessary for most users. We can adopt uv's documentation

```
The path to a `uv.toml` file to use for configuration.
While uv configuration can be included in a `pyproject.toml` file, it is not allowed in this context.
```

UV also has a `TY_CONFIG_FILE` environment variable. I'd be open to support this too (as simple as telling clap to respect the env variable)